actions/detect-version/src/main.ts

```diff
@@ -80,20 +82,50 @@ async function loadTagVersions(depth: number): Promise<void> {
 
   const knownVersions = await utils.getVersions()
 
-  const latestTag = utils.latestVersionTag(knownVersions)
+  let latestTag = utils.latestVersionTag(knownVersions)
   const latestSha = await git.resolveRef(latestTag)
-  const latestVersion = knownVersions[latestTag]
+  let latestVersion = knownVersions[latestTag]
 
-  const prevTag = await git.previousTag()
+  if (latestVersion && latestVersion.original) {
+    latestVersion.original = utils.sanitizeVersionInput(latestVersion.original)
+    latestVersion = version.parse(latestVersion.original)
+  }
+
+  if (latestTag.toLowerCase() === 'nightly') {
+    const sortedTags = utils.sortTagsBySemver(Object.keys(knownVersions))
+    const previousValidTag = sortedTags.find(
+      tag => tag.toLowerCase() !== 'nightly'
+    )
+    if (previousValidTag) {
+      latestTag = previousValidTag
+      latestVersion = knownVersions[previousValidTag]
+    }
+  }
+
+  let prevTag = await git.previousTag()
   const prevSha = await git.resolveRef(prevTag)
-  const prevVersion = knownVersions[prevTag]
+  let prevVersion = knownVersions[prevTag]
```

Comment:
Don't we need to sanitize prev version the same way we do for current and latest?